src/main/scala/com/campudus/tableaux/controller/TableauxController.scala

```diff
@@ -130,28 +71,45 @@ class TableauxController(override val config: TableauxConfig, override protected
   def deleteAttachment(tableId: TableId, columnId: ColumnId, rowId: RowId, uuid: String): Future[DomainObject] = {
     checkArguments(greaterZero(tableId), greaterZero(columnId), greaterZero(rowId), notNull(uuid, "uuid"))
     logger.info(s"deleteAttachment $tableId $columnId $rowId $uuid")
+
     //TODO introduce a Cell identifier with tableId, columnId and rowId
     repository.attachmentModel.delete(Attachment(tableId, columnId, rowId, UUID.fromString(uuid), None))
   }
 
-  private implicit def convertUnitToEmptyObject(unit: Future[Unit]): Future[EmptyObject] = {
-    unit map (s => EmptyObject())
-  }
+  def getCompleteTable(tableId: TableId): Future[CompleteTable] = {
+    checkArguments(greaterZero(tableId))
+    verticle.logger.info(s"getTable $tableId")
 
-  def changeTableName(tableId: TableId, tableName: String): Future[DomainObject] = {
-    checkArguments(greaterZero(tableId), notNull(tableName, "TableName"))
-    logger.info(s"changeTableName $tableId $tableName")
-    repository.changeTableName(tableId, tableName)
+    for {
+      table <- repository.getTable(tableId)
+      colList <- repository.getAllColumns(table.id)
+      rowList <- repository.getAllRows(table)
+    } yield CompleteTable(table, colList, rowList)
   }
 
-  def changeColumn(tableId: TableId,
-                   columnId: ColumnId,
-                   columnName: Option[String],
-                   ordering: Option[Ordering],
-                   kind: Option[TableauxDbType]): Future[DomainObject] = {
+  def createTable(tableName: String, columns: Seq[CreateColumn], rowsValues: Seq[Seq[_]]): Future[CompleteTable] = {
+    checkArguments(notNull(tableName, "TableName"), nonEmpty(columns, "columns"))
+    logger.info(s"createTable $tableName columns $rowsValues")
+
+    for {
+      table <- repository.createTable(tableName)
+      columns <- repository.createColumns(table.id, columns)
+      columnIds <- Future(columns.map(_.id))
+      rowsWithColumnIdAndValue <- Future.successful {
+        if (rowsValues.isEmpty) {
+          Seq()
+        } else {
+          rowsValues map {
+            columnIds.zip(_)
+          }
+        }
```

Comment:
Isn't `rowsValues map (columnIds.zip(_))` okay already? If it were empty before, you already have an empty `Seq()`.